src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -510,29 +511,54 @@ def process_engine_outputs(
         :param engine_outputs: the outputs from the engine
         :return: the output schema for the pipeline
         """
-        generated_tokens, generated_logits, finished_reason = engine_outputs
-        finished_reason = [f[0] for f in finished_reason]
-
-        sequences = self.tokenizer.batch_decode(
-            generated_tokens, skip_special_tokens=True
-        )
-        num_preds = kwargs.get("num_generated_predictions", 1)
-        prompts = kwargs.get("prompts")
 
         def _create_generated_text_output(
             sequence: str,
-            finish_reason: FinishReason,
+            finish_reason: FinishReason = None,
             logits: Optional[numpy.array] = None,
         ):
+            if finish_reason:
+                return GeneratedText(
+                    text=sequence,
+                    score=logits,
+                    finished=True,
+                    finished_reason=finish_reason.value,
+                )
             return GeneratedText(
                 text=sequence,
                 score=logits,
-                finished=True,
-                finished_reason=finish_reason.value,
+                finished=False,
             )
 
+        prompts = kwargs.get("prompts")
+
+        if self.streaming:
+            generations = []
+            for output in engine_outputs:
```

Comment:
nit: let's move this into a helper function `_stream_engine_outputs` or something to clean up the process outputs fn